fe/fe-core/src/main/java/com/starrocks/external/iceberg/cost/IcebergTableStatisticCalculator.java

```diff
@@ -96,16 +93,13 @@ private Statistics makeTableStatistics(List<Expression> icebergPredicates,
         LOG.debug("Begin to make iceberg table statistics!");
         List<Types.NestedField> columns = icebergTable.schema().columns();
         IcebergFileStats icebergFileStats = generateIcebergFileStats(icebergPredicates, columns);
-        if (icebergFileStats == null) {
-            return Statistics.builder().build();
-        }
 
         Map<Integer, String> idToColumnNames = columns.stream()
                 .filter(column -> column.type().isPrimitiveType())
                 .collect(Collectors.toMap(Types.NestedField::fieldId, column -> column.name()));
 
         Statistics.Builder statisticsBuilder = Statistics.builder();
-        double recordCount = Math.max(icebergFileStats.getRecordCount(), 1);
+        double recordCount = Math.max(icebergFileStats == null ? 0 : icebergFileStats.getRecordCount(), 1);
```

Comment:
what's the difference between NaN and 0? @Youngwb

Reply:
0 means no rows，NaN is not a legal state for row count